src/tools/compiletest/src/runtest.rs

```diff
@@ -2864,7 +2879,9 @@ impl<'test> TestCx<'test> {
             }
         }
 
-        let output_file = self.output_base_name().with_extension(kind);
+        let output_file = self.output_base_name()
```

Comment:
Why do we *not* want the stage here?

Reply:
Good question.  What are the arguments in favor of including the stage?  Is it to guard against using the wrong cached results if you run `x.py test --stage 1` and `x.py test --stage 2` separately?

One issue is that `update-references.sh` won't know which stage to use.  We could add that as another argument, and print the correct stage in the usage instructions.

Some of the other tests that don't include the stage may need to be updated as well.  I was reluctant to do so because I don't fully understand the need and consequences.

Reply:
> What are the arguments in favor of including the stage?

The way I always imagined it, we use the stage/target data to effectively "isolate" stage1 and stage2 tests from one another completely. To be honest, I think we should be using separate directories altogether (i.e, `src/test/stage1`). I'm not sure why we don't, I think for no particularly good reason.

But you're right that I don't know what can go wrong in *particular*.

Reply:
> One issue is that update-references.sh won't know which stage to use. We could add that as another argument, and print the correct stage in the usage instructions.

Seems fair. We can leave as is for now.

Reply:
If we moved to `./x.py bless`, then you could do `./x.py bless --stage 1` and so forth to specify the test suite that you ran. (Maybe it should be `./x.py test --bless`? Anyway.)